api/src/views/hardware_spec.rs

```diff
@@ -60,7 +67,16 @@ pub struct UpdateHardwareSpecView<'a> {
 impl<'a> View for UpdateHardwareSpecView<'a> {
     type Output = Option<models::HardwareSpec>;
 
-    fn execute(&self) -> ResponseResult<Self::Output> {
+    fn check_permissions(&self) -> ResponseResult<()> {
+        let user = self.context.user()?;
+        if user.has_permission(PermissionType::ModifyAllSpecs) {
+            Ok(())
+        } else {
+            Err(ClientError::PermissionDenied.into())
+        }
+    }
```

Comment:
Might not be worth it since its so simple but could we abstract this func a bit? Seems the same thing is used in a few of the views but with a different permission type.